docs/evm/guides/vrf.md

```diff
@@ -0,0 +1,288 @@
+
+**The Issue with Using `revertibleRandom()` Directly:**
+
+- When an untrusted party calls a contract function that uses `revertibleRandom()`, they receive the random number **during the transaction execution**.
+- **Post-selection** is the ability of the caller to abort the transaction if the random outcome is unfavorable. In this case, the user could choose to revert the transaction (for example, if they lose a bet) and attempt to call the function again in hopes of a better outcome.
+- This can lead to a form of **transaction reversion attack**, where the randomness can be exploited to repeatedly attempt transactions until a favorable result is obtained.
+
+**Example Attack Scenario**:
+
+Consider a decentralized coin toss using `revertibleRandom()`:
+
+- A malicious user could interact with the smart contract and immediately check whether they won.
+- If they lost the bet, they could programmatically revert the transaction before it is finalized.
+- This allows the user to retry and reattempt the random process multiple times until they win, undermining the fairness of the system.
+
+**Time-Locked Randomness with Commit-Reveal**
+
+The **commit-reveal scheme** solves the post-selection problem by splitting the random process into two phases:
+
+1. **Commit Phase**: The user commits to a hidden guess (via a hash of their guess and a nonce) **before** the randomness is revealed. This locks the user into their guess without revealing it to the smart contract or anyone else.
+2. **Reveal Phase**: After the randomness is securely generated and made public (e.g., using `revertibleRandom()` or another source), the user can then reveal their guess and nonce to complete the process.
+
+**Why This Is More Secure**
+
+- **No Post-Selection**: In the commit phase, the user cannot manipulate the random number because they are committed to their bet or action without knowing the outcome. Once they’ve committed, they are locked into their choice.
+- **Time Delay Between Commitment and Reveal**: By forcing the randomness to be generated after the commitment is made, users cannot use any immediate information about the random number to influence their decision.
+- **Immutable Commitment**: The use of a cryptographic hash (such as SHA-256) ensures that the user cannot change their guess after seeing the randomness. They must reveal the exact nonce and guess used during the commitment phase to verify the validity of their reveal.
+
+### **Commit-Reveal Scheme Example for a Random Coin Toss**
+
+Here is a Solidity implementation of a coin toss game using a secure commit-reveal scheme, where participants bet on the outcome (heads or tails), and randomness is fetched from the **Cadence Arch** contract.
+
+```solidity
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >=0.7.0 <0.9.0;
+
+contract SecureCoinToss {
+    address constant public cadenceArch = 0x0000000000000000000000010000000000000001;
+
+    struct Bet {
```

Comment:
Re: @tarakby's comment about gaming this contract

This struct will need to store the Flow block height at which the bet is made in conjunction with using `getRandomSource(uint64)` as a source of randomness